### ISSUE TITLE
Matching bug

### ISSUE THREAD

**Issue by Paikan** [JDonner/NoAho#3]
```python
# -*- coding: utf-8 -*-


from noaho import NoAho


a = NoAho()
a.add('an', 'an')
a.add('canal', 'canal')
a.add('e can oilfield', 'e can oilfield')

for match in a.findall_long('one canal'):
    print(match)

# matches with:
# (5, 7, 'an')
#
# expected match would be:
# (4, 9, 'canal')
```

**Comment by JDonner** [comment]
Ok. My making a 'long' version of Aho Corasick was kind of a hack, which
seemed to work - I wonder if these two bugs don't expose a design flaw.
I'll look at it. According to PyPi ~300 people a month download this,
yet I've gotten only your reports. So, many thanks!

On Wed, Feb 18, 2015, at 02:04 AM, Benjamin Devèze wrote:
> -*- coding: utf-8 -*-


> from noaho import NoAho


> a = NoAho()
>
a.add('an', 'an')
>
b.add('canal', 'canal')
>
c.add('e can oilfield', 'e can oilfield')

> for match in a.findall_long('one canal'):
>
    print(match)

> matches with:


> (5, 7, 'an')


> #


> expected match would be:


> (4, 9, 'canal')


> — Reply to this email directly or view it on GitHub[1].

>



Links:

  1. https://github.com/JDonner/NoAho/issues/3

**Comment by JDonner** [comment]
Ok, fixed, 0.9.6 is on github and PyPi. Thanks for the report!
